src/borg/testsuite/archiver.py

```diff
@@ -3765,6 +3765,64 @@ def test_change_location_does_not_change_algorithm_argon2(self):
+            borg_create_output = borg_create_output.split("\n")
+            borg_create_output.pop()  # The last line needs to be removed because its an empty string
+            borg_create_output = [line.split(":", 1) for line in borg_create_output]
+            borg_create_output = {
+                key: int(value) for key, value in borg_create_output if key in ("Added files", "Unchanged files", "Modified files")
+            }
+            return borg_create_output
+
+        # Test case set up: create a repository
+        self.cmd(f"--repo={self.repository_location}", "rcreate", KF_ENCRYPTION)
+        # Archive an empty dir
+        result = self.cmd(f"--repo={self.repository_location}", "create", "--stats", "test_archive", self.input_path)
+        result = to_dict(result)
+        assert result["Added files"] == 0
+        assert result["Unchanged files"] == 0
+        assert result["Modified files"] == 0
+        # Archive a dir with two added files
+        self.create_regular_file("testfile1", contents=b"test1")
+        self.create_regular_file("testfile2", contents=b"test2")
+        result = self.cmd(f"--repo={self.repository_location}", "create", "--stats", "test_archive2", self.input_path)
+        result = to_dict(result)
+        assert result["Added files"] == 2
+        assert result["Unchanged files"] == 0
+        assert result["Modified files"] == 0
+        # Archive a dir with 1 unmodified file and 1 modified
+        with open(self.input_path + "/testfile2", "a") as file:
+            file.write("data")
+        result = self.cmd(f"--repo={self.repository_location}", "create", "--stats", "test_archive3", self.input_path)
+        result = to_dict(result)
+        assert result["Added files"] == 1
+        assert result["Unchanged files"] == 1
+        assert result["Modified files"] == 0
+
+    def test_hashing_time(self):
+        """Tests hashing_time is positive or zero"""
+
+        def extract_hashing_time(borg_create_output: str) -> float:
+            borg_create_output = borg_create_output.split("\n")
+            borg_create_output.pop()  # The last line needs to be removed because its an empty string
+            borg_create_output = [line.split(":", 1) for line in borg_create_output]
+            hashing_time = [line for line in borg_create_output if line[0] == "Time spent in hashing"].pop()
+            hashing_time = hashing_time[1]
+            hashing_time = float(hashing_time[1][0 : len(hashing_time) - 7])
+            return hashing_time
+
+        # Test case set up: create a repository and a file
+        self.cmd(f"--repo={self.repository_location}", "rcreate", KF_ENCRYPTION)
+        self.create_regular_file("testfile1", contents=b"test1")
+        # Archive
+        result = self.cmd(f"--repo={self.repository_location}", "create", "--stats", "test_archive", self.input_path)
+        hashing_time = extract_hashing_time(result)
+
+        assert hashing_time >= 0.0
```

Comment:
this is a quite weak test considering you initialise the value to 0.0.

of course we can not know how long it takes, so guess the best we can do is check `> 0.0`.